src/engine/source/base/error.hpp

```diff
@@ -72,7 +72,19 @@ inline bool isError(const RespOrError<T>& respOrError)
  * @throws std::bad_variant_access if response is an error
  */
 template<typename T>
-inline T getResponse(RespOrError<T>&& response)
+inline auto getResponse(const RespOrError<T>& response) -> decltype(std::get<T>(response))
```

Comment:
Why are we using auto -> decltype(...) ?

I think it was more readable before.